core/dataflow/src/test/java/edu/uci/ics/texera/dataflow/source/twitter/TwitterConverterTest.java

```diff
@@ -0,0 +1,100 @@
+import edu.uci.ics.texera.dataflow.twitter.TwitterConverterConstants;
+import edu.uci.ics.texera.dataflow.twitter.TwitterConverterPredicate;
+import junit.framework.Assert;
+
+/**
+ * Test cases for operator {@link TwitterConverter}, 
+ *  which converts the JSON string representation of twitter to Texera fields.
+ * 
+ * @author Zuozhi Wang
+ *
+ */
+public class TwitterConverterTest {
+    
+    // get the sample twitter data from the perf test resources folder
+    public static String twitterFilePath =  Utils.getResourcePath("/sample-data-files/twitter/tweets.json", TexeraProject.TEXERA_PERFTEST).toString();
+
+    
+    public static List<Tuple> getAllSampleTwitterTuples() throws Exception {
+        
+        // read the JSON file into a list of JSON string tuples
+        JsonNode jsonNode = new ObjectMapper().readTree(new File(twitterFilePath));
+        ArrayList<Tuple> jsonStringTupleList = new ArrayList<>();
+        
+        Schema tupleSourceSchema = new Schema(
+                SchemaConstants._ID_ATTRIBUTE, 
+                new Attribute("twitterJson", AttributeType.STRING));
+        
+        for (JsonNode tweet : jsonNode) {
+            Tuple tuple = new Tuple(tupleSourceSchema, IDField.newRandomID(), new StringField(tweet.toString()));
+            jsonStringTupleList.add(tuple);
+        }
+        
+        // setup the twitter converter DAG
+        // TupleSource --> TwitterConverter --> TupleSink
+        TupleSourceOperator tupleSource = new TupleSourceOperator(jsonStringTupleList, tupleSourceSchema);
+
+        TwitterConverter twitterConverter = new TwitterConverterPredicate("twitterJson").newOperator();
+
+        TupleSink tupleSink = new TupleSinkPredicate(null, null).newOperator();
+        
+        twitterConverter.setInputOperator(tupleSource);
+        tupleSink.setInputOperator(twitterConverter);
+        
+        tupleSink.open();
+        
+        List<Tuple> tuples = tupleSink.collectAllTuples();
+        
+        tupleSink.close();
+        
+        return tuples;
+    }
+    
+    @Test
+    public void testTwitterConverter() throws Exception {
+        
+        List<Tuple> twitterTuples = getAllSampleTwitterTuples();
+        
+        Assert.assertTrue(twitterTuples.size() > 0);
+        
+        Tuple testTuple = twitterTuples.get(0);
+        
+        // make sure that all the additional attributes are in the schema
+        Assert.assertTrue(testTuple.getSchema().getAttributes().containsAll(TwitterConverterConstants.additionalAttributes));
+        
+        // make sure that all the tuple fields corresponds to the schema
+        for (Attribute attr : testTuple.getSchema().getAttributes()) {
+            Class<? extends IField> expectedFieldClass = attr.getType().getFieldClass();
+            Class<? extends IField> actualFieldClass = testTuple.getField(attr.getName()).getClass();
+
+            Assert.assertEquals(expectedFieldClass, actualFieldClass);
```

Comment:
Will the assertion in line 95 ever fail? Because both things are derived from testTuple.

Reply:
You are correct. This assertion will not fail. The `Tuple` class already checked that the actual data fields and the schema matches.  This unnecessary check is removed.